routing/router.go

```diff
@@ -1799,6 +1822,15 @@ func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, rt *route.Route) (
 	}
 
 	if failureCode != nil {
+		log.Debugf("Failure received for shard for payment %x: %v",
+			hash, outcome.paymentFailure)
+
+		// For MPP payments we don't mark them failed with the control
+		// tower, since we might want to try more shards.
```

Comment:
so in what cases _do_ we mark them failed??

Reply:
Not applicable anymore. Now we _always_ mark them failed.